jormungandr/src/main.rs

```diff
@@ -279,7 +281,23 @@ fn start_services(bootstrapped_node: BootstrappedNode) -> Result<(), start_up::E
         });
     }
 
-    services.spawn_try_future("sigint_listener", move |_info| ctrl_c().map_err(|_| ()));
+    services.spawn_try_future("sigint_listener", move |_info| {
+        signal::ctrl_c().map_err(|_| ())
+    });
+
+    #[cfg(unix)]
+    services.spawn_future("sigterm_listener", move |info| {
+        use signal::unix::SignalKind;
+
+        let signal = match signal::unix::signal(SignalKind::terminate()) {
+            Ok(signal) => signal,
+            Err(e) => {
+                warn!(info.logger(), "failed to install handler for SIGTERM"; "reason" => %e);
+                return future::ready(()).left_future();
+            }
+        };
+        signal.into_future().map(|_| ()).right_future()
+    });
```

Comment:
Just out of curiosity. What happens with other platforms?

Reply:
With what's available in tokio, "the other platforms" means Windows, and there we can only additionally handle Ctrl+Break. I'm not sure this is actually needed by anyone.

`SIGTERM` is the conventional way to gracefully stop a daemon on Unix, used by IOHK infrastructure as well.